dspace-uclouvain/src/main/java/org/dspace/uclouvain/pdfAttestationGenerator/xmlworkflow/actions/GeneratePDFAction.java

```diff
@@ -113,10 +113,14 @@ public List<String> getOptions() {
      * @param map The HashMap containing information about the submission
      * @return The resume as a String 
     */
-    private static String generateSubmissionResume(HashMap<String, List<String>> map) {
-        String title = map.get("dc_title").get(0);
-        List<String> authors = map.get("dc_contributor_author");
-        String abstractText = map.get("dc_description_abstract").get(0);
-        return "Title: " + title +"\nAuthor(s): " + authors.stream().reduce("", (subtotal, element) -> subtotal + element + "; ") +"\nAbstract: " + abstractText;
+    private static String generateSubmissionResume(HashMap<String, List<String>> map) throws ResumeGenerationException {
+        try {
+            String title = map.get("dc_title").get(0);
+            List<String> authors = map.get("dc_contributor_author");
+            String abstractText = map.get("dc_description_abstract").get(0);
+            return "Title: " + title +"\nAuthor(s): " + authors.stream().reduce("", (subtotal, element) -> subtotal + element + "; ") +"\nAbstract: " + abstractText;
+        } catch (Exception e) {
+            throw new ResumeGenerationException("Could not generate the mail submission resume due to a missing metadata");
```

Comment:
Append the original exception message to your exception ; Not need to have an explicit message if you use a specialized exception class
```java
throw new ResumeGenerationException("Submission mail generation failed :: "+e.getMessage());
```